foundation_page.py

```diff
@@ -0,0 +1,15 @@
+import time
+from utils import is_images_equal
+
+class FoundationPage():
+    def wait_page_stable(self, driver):
+        image_1 = driver.get_screenshot_as_png()
+        for x in range(5):
+            image_2 = driver.get_screenshot_as_png()
+            if is_images_equal(image_1, image_2):
+                break
+            time.sleep(0.01)
+            image_1 = driver.get_screenshot_as_png()
+
+    def open_page(self, driver, link):
```

Comment:
The same, as previous comment

Reply:
And I think, that me may avoid link parameter here too
But you have to rewrite code in this way:
Instead of link parameter use self.link
Before opening check, that class has this attribute, if not - raise custom exception like PageWithoutLinkException